go/mysql/conn.go

```diff
@@ -1474,30 +1474,40 @@ func (c *Conn) parseOKPacket(in []byte) (*PacketOK, error) {
 			if length == 0 {
 				return packetOK, nil
 			}
-			sscType, ok := data.readByte()
-			if !ok {
-				return fail("invalid OK packet session state change type: %v", sscType)
-			}
-			// If it's not a GTID, we don't care about it so we can return.
-			if sscType != SessionTrackGtids {
-				return packetOK, nil
-			}
 
-			// Move past the total length of the changed entity: 1 byte
-			_, ok = data.readByte()
-			if !ok {
-				return fail("invalid OK packet gtids length: %v", data)
-			}
-			// read (and ignore for now) the GTIDS encoding specification code: 1 byte
-			_, ok = data.readByte()
-			if !ok {
-				return fail("invalid OK packet gtids type: %v", data)
-			}
-			gtids, ok := data.readLenEncString()
-			if !ok {
-				return fail("invalid OK packet gtids: %v", data)
+			// Alright, now we need to read each sub packet from the session state change.
+			for data.pos < len(data.data) {
```

Comment:
I believe we can check for length > 0 here.

Reply:
It's what I tried first, but that fails since we're missing the nested inner bytes that encode the length of the inner strings. So that's why I moved away from that, but it looks like I didn't clean up the `length` stuff so I'll do that to simplify this.

Reply:
I have cleaned this up a bit and simplified how the loop terminates.